docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,306 @@
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "250m"
+          memory: "50Mi"
+        request:
+          cpu: "250m"
+          memory: "50Mi"
+    - name: step-restore
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/restorer
+      args:
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+    - name: step-build
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/builder
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -group=/layers/group.toml
+        - -plan=/layers/plan.toml
+      volumeMounts:
+        - name: layers-dir
+          mountPath: /layers
+      resources:
+        limits:
+          cpu: "500m"
+          memory: "1Gi"
+        request:
+          cpu: "500m"
+          memory: "1Gi"
+    - name: step-export
+      image: gcr.io/paketo-buildpacks/builder:latest
+      securityContext:
+        runAsUser: 1000
+      command:
+        - /cnb/lifecycle/exporter
+      args:
+        - -app=/workspace/source/$(build.source.contextDir)
+        - -layers=/layers
+        - -cache-dir=/cache
+        - -group=/layers/group.toml
+        - $(build.output.image)
+      volumeMounts:
+        - name: cache-dir
+          mountPath: /cache
+        - name: layers-dir
+          mountPath: /layers
+```
+
+### Risks and Mitigations
+
+Proper documentation needs to be made to communicate the existance of `flavours` for strategies of the same type to users. When it comes to UI, is the decision of the UI team to decide
+how they will expose(_interface_) this flavours to their end-users. Strategy admins have full responsability on the flavours they decide to define, but recommendations of flavours should be
+made for this repository.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- N/A
+
+##### Tech Preview -> GA
+
+- N/A
+
+**For non-optional features moving to GA, the graduation criteria must include end to end tests.**
+
+##### Removing a deprecated feature
+
+- N/A
+
+### Upgrade / Downgrade Strategy
+
+- N/A
+
+### Version Skew Strategy
+
+- N/A
+
+## Implementation History
+
+- N/A
+
+## Drawbacks
+
+None. Only that we will remove a feature that is currently not performing properly.
+
+## Alternatives
+
+Create a separate CRD for the container resources settings and allow the strategy to have an optional reference to it. This provides two potential benefits:
```

Comment:
Could you also add why this alternative didn't make it?

Reply:
Yes, but I can also explain it now. We need a proper proposal for this alternative. What we know so far is that we could have another CRD to manage the strategy resources, some of the open questions are:
- how to reference this CRD on the existing strategies?
- how can you define resources through this CRD for multiple steps on a single strategy. Similar to the issue we have now with the `spec.resources` in the Build, where we cannot define customize values per step. How can the CRD tackle it?
- impact on the user experience in terms of another CRD.
- would there be default resources, when the CRD is not referenced in the strategy?

Reply:
Thanks.